.github/workflows/lint.yml

```diff
@@ -0,0 +1,32 @@
+name: Go
+
+on: [push, pull_request]
+
+jobs:
+
+  build:
+
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: ['1.14', '1.15']
+
+    steps:
+
+    - name: Set up Go ${{ matrix.go-version }}
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: go mod download
+
+    - name: Download Linter
+      run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.34.1
+
+    - name: Lint
+      run: golangci-lint run
```

Comment:
Why not simply adding these to the main `build.yml` befre build and test steps?

Seems a bit of a waste replicating the whole GitHub Action! 😕

Reply:
This will cause the tests not to run at all if linting fails. As far as I can remember this was the reason to separate those two, so in case when linting fails the tests and code coverage are still checked.

Reply:
Yeah, I was more questioning the rationale because I've seen other projects where the lint is just a step of the build.

If the only concern is not failing the whole job we could set `continue-on-error: true` (see [GitHub Action doc](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error)).

What do you think?

Reply:
As far as I understand `continue-on-error: true` will make the whole job pass despite lint errors, which I don't think is what we want. In this case I think it's better to make it a step of the build. The only benefit of having it as a separate job is that than in the PR you can immediately see:

- Tests passed
- Lint passed
- Coverage check passed (btw I have noticed it's not here, maybe the token env variable has not been set when migrating to GitHub actions? if you could take a look)

as opposed to 
- Tests **and** lint passed
- Coverage check passed

so when making a PR with both tests failing and lint errors you can see that and fix both before pushing a new version instead of first fixing tests/lint (depending on which one is defined first in the job), pushing again just to see that the other one has also failed.

However since the repo is not huge and tests take only few minutes to run this is not a big issue and. So let me know if you think it's better to have it in one job and I'll put it after the tests in the main job without `continue-on-error` flag.

Reply:
> As far as I understand continue-on-error: true will make the whole job pass despite lint errors, which I don't think is what we want. 

I was hoping we could see at least a warning or additional indicator next to the green tick in case of lint failure but apparently that's not the case (see [feature discussion](https://github.community/t/continue-on-error-allow-failure-ui-indication/16773)).

Let's keep it separate then! :)